noir_stdlib/src/mem.nr

```diff
@@ -4,3 +4,14 @@
 #[builtin(zeroed)]
 pub fn zeroed<T>() -> T {}
 
+/// Transmutes a value of type T to a value of type U.
+///
+/// This operation is checked in that in a later stage of compilation
```

Comment:
`This operation is checked in that in a later stage`

Awkward phrasing, perhaps something like:

`This operation is checked in a stage of compilation after normal type checking, where`

?

Reply:
Reworded this a bit